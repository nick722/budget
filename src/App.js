import React from "react";
import Transactions from "./components/transactions/transactions";
import Header from "./components/header/Header";
import Footer from "./components/footer/Footer";

import "./App.scss";

function App() {
  return (
    <>
      <Header />
      <Transactions />
      <Footer />
    </>
  );
}

export default App;
