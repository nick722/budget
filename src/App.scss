$border-color: #d7d7d7;

body {
  height: 100vh;
  margin: 0;
  padding: 0;
}

#root {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header" "main" "footer";
  grid-template-rows: 4rem auto 3rem;
  max-inline-size: 1200px;
  min-block-size: 100vh;
  margin: auto;
  border-left: 1px solid $border-color;
  border-right: 1px solid $border-color;
}
