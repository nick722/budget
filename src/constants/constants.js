export const PATHNAMES = {
  HOME: "/",
  CATEGORIES: "/categories"
};
