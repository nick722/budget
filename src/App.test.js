import React from "react";
import { MemoryRouter } from "react-router-dom";
import { Provider as ReduxProvider } from "react-redux";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import createStore from "./store/create-store";
import App from "./App";
import Transactions from "./components/transactions/transactions";
import Categories from "./components/categories/categories";
import { PATHNAMES } from "./constants/constants";

const appJsx = (
  <ReduxProvider store={createStore()}>
    <MemoryRouter>
      <App />
    </MemoryRouter>
  </ReduxProvider>
);

describe("App", () => {
  let component;

  beforeEach(() => {
    component = shallow(<App />);
  });

  it("should render and match snapshot", () => {
    const tree = renderer.create(appJsx).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Switch", () => {
    let pageSwitch;
    let pageRoutes;

    beforeEach(() => {
      pageSwitch = component.find("Switch");
      pageRoutes = pageSwitch.find("Route");
    });

    it("should render a Switch", () => {
      expect(pageSwitch).toHaveLength(1);
    });

    it("should render 2 routes", () => {
      expect(pageRoutes).toHaveLength(2);
    });

    describe("Transactions Route", () => {
      let transactionsRoute;

      beforeEach(() => {
        transactionsRoute = pageRoutes.find({ component: Transactions });
      });

      it("should render route for Transactions component", () => {
        expect(transactionsRoute).toHaveLength(1);
      });

      it("should have correct path", () => {
        expect(transactionsRoute.prop("path")).toEqual(PATHNAMES.HOME);
      });

      it("should have 'exact' prop passed", () => {
        expect(transactionsRoute.prop("exact")).toBeTruthy();
      });
    });

    describe("Categories Route", () => {
      let categoriesRoute;

      beforeEach(() => {
        categoriesRoute = pageRoutes.find({
          component: Categories
        });
      });

      it("should render route for Categories component", () => {
        expect(categoriesRoute).toHaveLength(1);
      });

      it("should have correct path", () => {
        expect(categoriesRoute.prop("path")).toEqual(PATHNAMES.CATEGORIES);
      });
    });
  });
});
