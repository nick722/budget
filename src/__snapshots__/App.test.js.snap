// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Categories page WHEN Categories link is clicked 1`] = `
Array [
  <header>
    <div
      className="logo"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-wallet fa-w-16 "
        data-icon="wallet"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <nav
      className="site-navigation"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/categories"
        onClick={[Function]}
      >
        Categories
      </a>
    </nav>
  </header>,
  <div
    className="transactions"
  >
    <div
      className="transactions__nave-pane"
    >
      <div
        className="transactions__month-panel"
      >
        <div>
          <button
            className="transactions__month-panel--arrow"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-left fa-w-10 "
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
        <div
          className="transactions__month-panel--banner"
        >
          October 2019
        </div>
        <div>
          <button
            className="transactions__month-panel--arrow"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="balance-pane"
    >
      <div
        className="balance-banner"
      >
        <span>
          Overall balance 
        </span>
        <span
          className="balance-banner__sum"
        >
          5100 rub.
        </span>
      </div>
    </div>
    <div
      className="transactions-list"
    >
      <table>
        <thead>
          <tr>
            <th>
              Description
            </th>
            <th>
              Amount
            </th>
          </tr>
        </thead>
        <tbody
          className="transactions-entries"
        />
      </table>
    </div>
    <div
      className="instruments-pane"
    >
      <button
        className="add-button"
        onClick={[Function]}
      >
        Add transaction
      </button>
    </div>
  </div>,
  <footer>
    © Copyright 2019
  </footer>,
]
`;

exports[`App should render Transactions Component WHEN Home link is clicked 1`] = `
Array [
  <header>
    <div
      className="logo"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-wallet fa-w-16 "
        data-icon="wallet"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <nav
      className="site-navigation"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/categories"
        onClick={[Function]}
      >
        Categories
      </a>
    </nav>
  </header>,
  <div
    className="transactions"
  >
    <div
      className="transactions__nave-pane"
    >
      <div
        className="transactions__month-panel"
      >
        <div>
          <button
            className="transactions__month-panel--arrow"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-left fa-w-10 "
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
        <div
          className="transactions__month-panel--banner"
        >
          October 2019
        </div>
        <div>
          <button
            className="transactions__month-panel--arrow"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="balance-pane"
    >
      <div
        className="balance-banner"
      >
        <span>
          Overall balance 
        </span>
        <span
          className="balance-banner__sum"
        >
          5100 rub.
        </span>
      </div>
    </div>
    <div
      className="transactions-list"
    >
      <table>
        <thead>
          <tr>
            <th>
              Description
            </th>
            <th>
              Amount
            </th>
          </tr>
        </thead>
        <tbody
          className="transactions-entries"
        />
      </table>
    </div>
    <div
      className="instruments-pane"
    >
      <button
        className="add-button"
        onClick={[Function]}
      >
        Add transaction
      </button>
    </div>
  </div>,
  <footer>
    © Copyright 2019
  </footer>,
]
`;

exports[`App should render and match snapshot 1`] = `
Array [
  <header>
    <div
      className="logo"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-wallet fa-w-16 "
        data-icon="wallet"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <nav
      className="site-navigation"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/categories"
        onClick={[Function]}
      >
        Categories
      </a>
    </nav>
  </header>,
  <div
    className="transactions"
  >
    <div
      className="transactions__nave-pane"
    >
      <div
        className="transactions__month-panel"
      >
        <div>
          <button
            className="transactions__month-panel--arrow"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-left fa-w-10 "
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
        <div
          className="transactions__month-panel--banner"
        >
          October 2019
        </div>
        <div>
          <button
            className="transactions__month-panel--arrow"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="balance-pane"
    >
      <div
        className="balance-banner"
      >
        <span>
          Overall balance 
        </span>
        <span
          className="balance-banner__sum"
        >
          5100 rub.
        </span>
      </div>
    </div>
    <div
      className="transactions-list"
    >
      <table>
        <thead>
          <tr>
            <th>
              Description
            </th>
            <th>
              Amount
            </th>
          </tr>
        </thead>
        <tbody
          className="transactions-entries"
        />
      </table>
    </div>
    <div
      className="instruments-pane"
    >
      <button
        className="add-button"
        onClick={[Function]}
      >
        Add transaction
      </button>
    </div>
  </div>,
  <footer>
    © Copyright 2019
  </footer>,
]
`;
